src/ncp/ncp_base_mtd.cpp

```diff
@@ -2619,6 +2619,118 @@ template <> otError NcpBase::HandlePropertyGet<SPINEL_PROP_THREAD_NETWORK_TIME>(
 }
 #endif // OPENTHREAD_CONFIG_ENABLE_TIME_SYNC
 
+#if OPENTHREAD_CONFIG_ENABLE_TIME_SYNC && OPENTHREAD_CONFIG_ENABLE_PERFORMANCE_TEST
+template <> otError NcpBase::HandlePropertySet<SPINEL_PROP_PERFORMANCE_LATENCY_TEST>(void)
+{
+    otError             error = OT_ERROR_NONE;
+    const otIp6Address *peerAddr;
+    uint16_t            length;
+    bool                isSender;
+    otMessage *         message = NULL;
+    otMessageInfo       messageInfo;
+    char                payload[1500];
+
+    SuccessOrExit(error = mDecoder.ReadIp6Address(peerAddr));
+    SuccessOrExit(error = mDecoder.ReadUint16(length));
+    SuccessOrExit(error = mDecoder.ReadBool(isSender));
+
+    memset(&mSocket, 0, sizeof(mSocket));
+
+    error = otUdpClose(&mSocket);
+    SuccessOrExit(error);
+
+    error = otUdpOpen(mInstance, &mSocket, HandleUdpReceive, this);
+    SuccessOrExit(error);
+
+    {
+        otSockAddr sockAddr;
+        memset(&sockAddr, 0, sizeof(sockAddr));
+        error = otIp6AddressFromString("::", &sockAddr.mAddress);
+        SuccessOrExit(error);
+        sockAddr.mPort    = 1234;
+        sockAddr.mScopeId = OT_NETIF_INTERFACE_ID_THREAD;
+        error             = otUdpBind(&mSocket, &sockAddr);
+        SuccessOrExit(error);
+    }
+
+    if (isSender)
+    {
+        memset(&messageInfo, 0, sizeof(messageInfo));
+        memcpy(&messageInfo.mPeerAddr, peerAddr, sizeof(otIp6Address));
+        messageInfo.mPeerPort    = 1234;
+        messageInfo.mInterfaceId = OT_NETIF_INTERFACE_ID_THREAD;
+
+        message = otUdpNewMessage(mInstance, true);
+        VerifyOrExit(message != NULL, error = OT_ERROR_NO_BUFS);
+
+        {
+            otNetworkTimeStatus status;
+            uint64_t            txTimestamp;
+            status = otNetworkTimeGet(mInstance, txTimestamp);
+            if (status != OT_NETWORK_TIME_SYNCHRONIZED)
+            {
+                txTimestamp = 0;
+            }
+            memset(payload, 'T', (size_t)length);
+            memcpy(payload, &txTimestamp, sizeof(uint64_t));
+
+            error = otMessageAppend(message, payload, static_cast<uint16_t>(length));
+            SuccessOrExit(error);
+        }
+
+        error = otUdpSend(&mSocket, message, &messageInfo);
+        SuccessOrExit(error);
+    }
+
+exit:
+    if (error != OT_ERROR_NONE && message != NULL)
+    {
+        otMessageFree(message);
+    }
+    return error;
+}
+
+template <> otError NcpBase::HandlePropertyGet<SPINEL_PROP_PERFORMANCE_LATENCY>(void)
+{
+    otError error = mEncoder.WriteUint32(mLatency);
+    mLatency      = 0;
```

Comment:
How about setting it to zero from start of test in `HandlePropertySet<SPINEL_PROP_PERFORMANCE_LATENCY_TEST>(void)`?

Note that the `HandlePropertyGet<SPINEL_PROP_PERFORMANCE_LATENCY>(void)` can called multiple times to prepare the response (in case there is no NCP buffer available), so would be better not to clear the value.

Reply:
Thanks a lot for your suggestion. I think it is hard to check the UDP packet is accept or not in the transmission test if we put the 'clear' operation in the start of test . The property is only got from the receiving side, if the value is zero after sending the UDP packet, which means the  packet is lost from the receiving side. The function named `HandlePropertyGet<SPINEL_PROP_PERFORMANCE_LATENCY>(void)` is called once for the receiving side.

Reply:
@abtink, I got it. Thanks a lot for your suggestion.

Reply:
I added `HandlePropertySet` to clear the value.